### PR TITLE
LIME-406: Add notification goal completed and new achievement

### LINKED ISSUE #401: Add notification about new achievement
When User made a new Achievement he should be notified about it in Notifications in Header component

Image

Text message can be for example: **"You have reached a new achievement {achievement_name}!"** or **Congratulation with new achievement {achievement_name}!**

### LINKED ISSUE #406: Add notification about goal fulfillment
When User complete a Goal he should be notified about it in Notifications in Header component

Image

Text message can be for example: "Congratulation with the completed goal {goal_name}!

### PR DESCRIPTION
**Pull request by valery-chumak**
https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/72942456/e18b410a-c53e-4d98-b6a0-3c793e3285bf

### CODE REVIEW

File: backend/src/common/services/calculation-progress/calculation-progress.service.ts [hunk 1]
```diff
@@ -20,15 +21,20 @@ class CalculationProgressService {
     private achievementService: AchievementService;
     private goalService: GoalService;
     private userAchievementsService: UserAchievementService;
+    private notificationService: NotificationService;
 
+    /* eslint-disable max-params */
     public constructor(
+        /* eslint-enable max-params */
```
Comment (stpalamar) [inline backend/src/common/services/calculation-progress/calculation-progress.service.ts]: why do we need to disable linter?

Comment (stpalamar) [reply backend/src/common/services/calculation-progress/calculation-progress.service.ts]: and enable right after one line))

Comment (valery-chumak) [reply backend/src/common/services/calculation-progress/calculation-progress.service.ts]: @stpalamar linter does not allow me to pass more than 3 params to constructor, but I need to add Notification Service to it and 3 others are needed for calculation which Serhii implemented.
If you know how to make it without linter disabling, I would be very thankful))

Comment (jhladkov) [reply backend/src/common/services/calculation-progress/calculation-progress.service.ts]: you can pass an object to constructor

Comment (jhladkov) [reply backend/src/common/services/calculation-progress/calculation-progress.service.ts]: Here an example:
https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/pull/387/files#diff-33831ca28b49ebcc1f5b505ce580b4d87cdc3caf7cb4dcbfebee441cf6497aaeR14

File: backend/src/common/services/calculation-progress/calculation-progress.service.ts [hunk 2]
```diff
@@ -62,6 +73,25 @@ class CalculationProgressService {
...
+                hasCompletedGoal = true;
+            }
+        }
+
+        if (hasCompletedGoal) {
+            await this.notificationService.create({
+                userId,
+                title: 'Goal Completed',
+                message:
+                    'Congratulations! You have completed your goal. Go to Goals Page to see your results.',
```
Comment (jhladkov) [inline backend/src/common/services/calculation-progress/calculation-progress.service.ts]: Maybe it's better to save messages in enum. What do you think?

Comment (valery-chumak) [reply backend/src/common/services/calculation-progress/calculation-progress.service.ts]: Fixed)

**Review by jhladkov — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**